### ISSUE TITLE
Select interaction: Selected feature not on top in OL6

### ISSUE THREAD

**Issue by JackD83** [openlayers/openlayers#10227]
**Describe the bug**
In ol5 (5.3.1), a feature that was selected using the select interaction was shown on top of any other feature in any other layer. 
In ol6 (6.1.0), the selected feature stays below other features. 

Just changing the zIndex of the feature is no workaround, as the other icons can be in other layers.

![ol6](https://user-images.githubusercontent.com/15889805/67851269-64267f00-fb0a-11e9-8b14-95b05e785b2e.png)

**To Reproduce**
See code example [here](https://codesandbox.io/s/icon-e99bh?fontsize=14)
Change package.json "ol" to 5.3.1 to see old an desired behavior. 

**Expected behavior**
The selected feature should be on top to avoid confusion in cluttered spaces with lots of overlaying features

**Comment by jahow** [comment]
The select interaction behaviour has indeed been simplified and does not use an overlay by default anymore. If you have several layers and you want to have your selected feature on top, you'd have to create an overlay yourself and use the select interaction features collection as source:
```js
let overlay = new VectorLayer({
  style: hoverStyle,
  source: new VectorSource({
    features: selectInteraction.getFeatures()
  }),
  zIndex: 100
});
```

**Comment by JackD83** [comment]
Thank you! This resolves this issue but spawned #10234
